src/code/sys_matrix.c

```diff
@@ -603,16 +603,30 @@ Mtx* Matrix_MtxFToMtx(MtxF* src, Mtx* dest) {
     return dest;
 }
 
+#ifdef OOT_DEBUG
+
 Mtx* Matrix_ToMtx(Mtx* dest, char* file, s32 line) {
-    return Matrix_MtxFToMtx(Matrix_CheckFloats(sCurrentMatrix, file, line), dest);
+    return Matrix_MtxFToMtx(MATRIX_CHECK_FLOATS(sCurrentMatrix, file, line), dest);
 }
 
 Mtx* Matrix_NewMtx(GraphicsContext* gfxCtx, char* file, s32 line) {
-    return Matrix_ToMtx(Graph_Alloc(gfxCtx, sizeof(Mtx)), file, line);
+    return Matrix_ToMtx(GRAPH_ALLOC(gfxCtx, sizeof(Mtx)), file, line);
+}
+
+#else
+
+Mtx* Matrix_ToMtx(Mtx* dest) {
+    return Matrix_MtxFToMtx(sCurrentMatrix, dest);
 }
 
+Mtx* Matrix_NewMtx(GraphicsContext* gfxCtx) {
+    return Matrix_ToMtx(GRAPH_ALLOC(gfxCtx, sizeof(Mtx)));
+}
+
+#endif /* OOT_DEBUG */
```

Comment:
Sorry, "add new functions" was a poor choice of words on my part. I meant whether the we should name the debug and retail versions the same (as if they were "modified" for debug) or if they should have separate names (as if they were "new" functions for debug).

I don't think it really makes sense to split this change into another PR, because doing something like `Matrix_ToMtxDebug(..., "", 0)` for the retail version in the meantime just seems obviously broken, and it's not a large change to keep it here, and I've already made the change here.

Reply:
basically the way it is currently doesn't address my comment or anghelo's (?) from the previous thread

I personally feel very strongly about not adding code to the repo that doesn't represent the matching game

Reply:
Yeah sorry, I misunderstood before.

I disagree for exactly the same reason, the non-debug versions in this PR do match retail OOT versions, while `Matrix_ToMtxDebug(..., "", 0)` doesn't match anything at all

Reply:
Oh I didn't know they match retail, I assumed you just winged it. Then okay. I have no easy way to check if they do match so I'll just trust you on it

Just one more thing, anghelo said

> Those functions are swapped between debug and non debug

I understood it as "they are in a different order", but in this PR they're in the same order?

Reply:
I think he meant that the debug versions completely replaced the non-debug versions in the ROM so it looks like they've been "swapped out". I don't think we can know what order they appeared in the original source

Reply:
as far as we know the order in the original source is the order they appear in the rom so I use "order" for both

ok so I misunderstood then, thanks all's good